articles/tests/test_views.py

```diff
@@ -260,3 +263,186 @@ def test_get_article_with_invalid_slug_fail(self) -> None:
             **self.bearer_token,
         )
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_like_article_pass(self) -> None:
+        """test the ability to like an article"""
+        likes = self.article.likes.count()
+        response = self.client.patch(
+            reverse("article-favorite", kwargs={"slug": self.article.slug}),
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data.get("likes")), likes + 1)
+
+    def test_unlike_article_pass(self) -> None:
+        """test the ability to unlike an article that is already liked. Removes user from the likes list"""
+        likes = self.article.likes.count()
+        response = self.client.patch(
+            reverse("article-favorite", kwargs={"slug": self.article.slug}),
+            **self.bearer_token,
+        )
+        response = self.client.patch(
+            reverse("article-favorite", kwargs={"slug": self.article.slug}),
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data.get("likes")), likes)
```

Comment:
if they have unliked, shouldn't the likes reduce by 1?

Reply:
If you like an article then unlike it then the number of likes remains the same.